docs/.pre-commit-config.yaml

```
@@ -0,0 +1,46 @@
+---
+repos:
+  # import formatter with black configurations
+    - repo: https://github.com/pycqa/isort
+      rev: 5.10.1
```

I noticed a number of these dependencies might reference non-current revisions, for example this one. Consider using `pre-commit autoupdate` to automatically update these to the latest.
```suggestion
      rev: 5.11.4
```